### PR TITLE
Fix interact_bls() for lightcurve v2 compatibility

### LINKED ISSUE #848: Lightkurve2 - interact_bls() broken
#### Problem description

`interact_bls()` is not yet working in the current `master` (Sep 20, 2020), due to the change of making time, flux from unitless numbers to `Time` / `Quantity`: 

```
ERROR:tornado.application:Uncaught exception GET /autoload.js?bokeh-autoload-element=1001&bokeh-absolute-url=http://localhost:53003&resources=none (::1)
...
site-packages\astropy\time\core.py", line 1734, in __getattr__
    return self.__getattribute__(attr)
AttributeError: 'TimeDelta' object has no attribute 'log10'
...
  File "c:\dev\lightkurve\lightkurve\interact_bls.py", line 449, in _create_interact_ui
    period_values = np.logspace(np.log10(minp), np.log10(maxp), npoints_slider.value)
TypeError: loop of ufunc does not support argument 0 of type TimeDelta which has no callable log10 method
```

I fixed a couple of those issues locally (by stripping units) and then stopped. 
Maybe there is a more robust way to deal with it, rather than converting `Quantity` / `Time` back to unitless number upon encountering an error?

#### Environment

-  platform (e.g. Linux, OSX, Windows): Windows
-  lightkurve version (e.g. 1.0b6): `master`, lightkurve2 alpha based
-  installation method (e.g. pip, conda, source): soruce

### PR DESCRIPTION
**Pull request by orionlee**
Closes #848 

- [x] Make initial display work
- [x] Make interaction work
- [ ] (PENDIG review discussion) Fix `model_lc` boundaries 
- [x] Fix folded lightcurve does not appear to have cover the entire phase.
- [x] Fix codes causing deprecation warnings.

### CODE REVIEW

**Comment by barentsen** [comment]
Wow, it looks like the migration to `TimeSeries` really did break `interact_bls()` in a significant way. Thank you so much for figuring this out.

I will wait to review & merge until you indicate this PR is ready.  :+1:

**Comment by orionlee** [comment]
The fix is largely done aside for a few specific issues that I'll need to look into further, but they shouldn't cause any significant changes.

For the overall approach, I tried to keep the unit around and strip it when necessary. It ends up somewhat complicated, with interop issues with both numpy and bokeh. It makes me wonder if we should try to create a unitless wrapper for lightcurve and let the code use it instead.

File: lightkurve/interact_bls.py [hunk 1]
```diff
@@ -459,24 +492,28 @@ def _create_interact_ui(doc, minp=minimum_period, maxp=maximum_period, resolutio
...
         # Set up the model LC
         mf = model.model(lc.time, best_period, duration_slider.value, best_t0)
         mf /= np.median(mf)
         mask = ~(convolve(np.asarray(mf == np.median(mf)), Box1DKernel(2)) > 0.9)
         model_lc = LightCurve(lc.time[mask], mf[mask])
-        model_lc = model_lc.append(LightCurve([(lc.time[0] - best_t0) + best_period/2], [1]))
-        model_lc = model_lc.append(LightCurve([(lc.time[0] - best_t0) + 3*best_period/2], [1]))
+        # Need to use raw value for best_t0 and best_period so that the result is of type Time, rather than TimeDelta
+        model_lc = model_lc.append(LightCurve(((lc.time[0] - best_t0.value) + best_period.value/2).reshape((1,)), [1]))
+        model_lc = model_lc.append(LightCurve(((lc.time[0] - best_t0.value) + 3*best_period.value/2).reshape((1,)), [1]))
```
[the remark this answers is not in the document]
Comment (orionlee) [reply lightkurve/interact_bls.py]: As it is, the 2 added points to `model_lc` skews the time range shown in lightcurve, making it not very helpful.

image

**Comment by orionlee** [comment]
The code changes are complete.
- I'd like to get some feedback on the `model_lc` code
- there are some minor enhancement I'd like to add. See issue #854 . If agreed, I can add them to this PR as the changes are quite small and isolated.
- The codes could use some re-factoring, but I don't want to add it to the PR, adding complexity to it.

**Comment by orionlee** [comment]
For `interact_bls()` in Lightkurve2, there is a separate issue on performance.
As it is, `interact_bls()` in lkv2 code base is noticeably slower than lkv1 equivalent.

On my machine, running `interact_bls()` on a Kepler target from a quarter (~4400 data points),
- lightkurve 1: 2-3 seconds
- lightkurve 2: 20 seconds

The issue should be strictly with lightkurve. In my test, both lightkurve 1 and 2 use the same set of dependent libraries, e.g., Astropy 4.1rc1.

**Comment by orionlee** [comment]
Rebased against the latest `master`.
@barentsen The PR is ready to be reviewed / merged: the `model_lc` issue I raised is really about an existing issue (bug I believed) that is not strictly necessary for the v2 migration. It can be dealt with separately if so desired.

**Review by barentsen — APPROVED** [review]
Thanks @orionlee! :+1:

**Pull request merged**